src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,194 @@
+
+
+def get_star_catalog(catalog, magnitude_cutoff=8.0, row_limit=1000000):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalogues. Default: Yale
+    min_magnitude: float
+        Minimum value for magnitude used in lookup
+    max_magnitude: float
+        Maximum value for magnitude used in lookup
+    row_limit: int
+        Maximum number of rows for the star catalog lookup
+
+    Returns
+    ----------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
+    """
+    from astroquery.vizier import Vizier
+
+    catalog_dict = StarCatalogues[catalog].value
+
+    vizier = Vizier(
+        catalog=catalog_dict["directory"],
+        columns=catalog_dict["columns"],
+        row_limit=row_limit,
+    )
+
+    stars = vizier.query_constraints(Vmag=f"<{magnitude_cutoff}")[0]
+
+    stars.meta["Catalog"] = StarCatalogues[catalog].value
+
+    return stars
+
+
+def get_bright_stars(
+    time, catalog="Yale", pointing=None, radius=None, magnitude_cut=None
+):
+    """
+    Get an astropy table of bright stars from the Yale bright star catalog
+
+    Parameters
+    ----------
+    time: astropy Time
+        time to which proper motion is applied
+    catalog : string
+        name of the catalog to be used available catalogues are 'Yale' and 'Hipparcos'. Default: 'Yale'
+    pointing: astropy Skycoord
+       pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
+       Radius of the sky region around pointing position. Default: full sky
+    magnitude_cut: float
+        Return only stars above a given absolute magnitude. Default: None (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
+    """
+
+    from importlib.resources import files
+
+    cat = StarCatalogues[catalog].value
+    record = cat["record"]
+
+    f = files("ctapipe").joinpath(f"resources/{record}.fits.gz")
+    stars = Table.read(f)
+
+    stars["ra_dec"] = SkyCoord(
+        ra=Angle(
+            stars[cat["coordinates"]["RA"]["column"]],
+            unit=u.Unit(cat["coordinates"]["RA"]["unit"]),
+        ),
+        dec=Angle(
+            stars[cat["coordinates"]["DE"]["column"]],
+            unit=u.Unit(cat["coordinates"]["DE"]["unit"]),
+        ),
+        pm_ra_cosdec=stars[cat["coordinates"]["pmRA"]["column"]],
+        pm_dec=stars[cat["coordinates"]["pmDE"]["column"]],
+        frame=cat["coordinates"]["frame"],
+        obstime=Time(cat["coordinates"]["epoch"]),
+    )
+    stars["ra_dec"] = stars["ra_dec"].apply_space_motion(new_obstime=time)
+    stars["ra_dec"] = SkyCoord(
```

Comment:
Please use the more appropriate workaround given by the astropy maintainers of using the `UnitSphericalRepresentation`